docs/docs/guides/smart_contracts/smart_contracts_guide.md

```diff
@@ -52,384 +52,353 @@ npm init -y
 
 ```bash
 node compile.js
 ```
 
-If everything is working correctly, you should see both the Contract Bytecode and the Contract ABI logged to the console.
+If everything is working correctly, you should see the contract's bytecode and ABI logged to the console and see two new files (`MyContractAbi.json` and `MyContractBytecode.bin`) in the project directory.
 
 :::tip
-📝 There are couple of other ways to get the bytecode and the ABI like using Remix and check the _Compilation Details_ after compiling the smart contract (https://remix-ide.readthedocs.io/en/latest/run.html#using-the-abi-with-ataddress).
+Alternatively, you can generate the bytecode and ABI with `npm i solc && npx solcjs MyContract.sol --bin --abi`. Keep in mind that this will generate files with different names than those used in this tutorial, so the files would need to be renamed to `MyContractBytecode.bin` and `MyContractAbi.json` to follow the rest of this guide exactly as it's written. Learn more about `solc-js` at https://github.com/ethereum/solc-js.
 :::
 
-## Step 5: Set up web3.js and connect to the Ganache network
+## Step 5: Set Up Web3.js and Hardhat
+
+In this step, we will set up Web3.js and Hardhat.
+
+First, install the required packages with npm:
 
-In this step, we will set up the web3.js library and connect to the Ganache network. So, be sure to run Ganache if you did not already did.
+```bash
+npm i web3 hardhat @nomicfoundation/hardhat-web3-v4
+```
 
-First, install the `web3` package using npm:
+Next, initialize the Hardhat project:
 
 ```bash
-npm i web3
+npx hardhat init
+```
+
+Initializing the Hardhat project will require responding to several prompts - select the default option for each prompt.
+
+After you initialize the Hardhat project, a number of new files and directories will be created, including a Hardhat configuration file called `hardhat.config.js`. Edit `hardhat.config.js` to include the [Web3.js plugin for Hardhat](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-web3-v4).
+
+```js
+require("@nomicfoundation/hardhat-toolbox");
+require("@nomicfoundation/hardhat-web3-v4"); // <=== add this import
+
+/** @type import('hardhat/config').HardhatUserConfig */
+module.exports = {
+  solidity: "0.8.24",
+};
+```
+
+To start the Hardhat development network, execute the following command:
+
+```bash
+npx hardhat node
+```
+
+Executing this command will produce the following output, which provides the URL that can be used to connect to the development network as well as the development network's test accounts:
+
+```bash
+Started HTTP and WebSocket JSON-RPC server at http://127.0.0.1:8545/
+
+Accounts
+========
+
+WARNING: These accounts, and their private keys, are publicly known.
+Any funds sent to them on Mainnet or any other live network WILL BE LOST.
+
+Account #0: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 (10000 ETH)
+Private Key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```

Comment:
Usually when we deal with docs and private keys, we want to hide the actual private keys in case theres a beginner who decides to reuse an exposed key
```suggestion
Private Key: privateKey
```

Reply:
Got it...thank you!